src/__tests__/pyunicoreWidget.test.tsx

```diff
@@ -0,0 +1,145 @@
+// mock the handler for api requests
+jest.mock('../handler', () => {
+  return {
+    __esModule: true,
+    requestAPI: jest
+      .fn()
+      .mockImplementation((url, init, settings) => Promise.resolve(data))
+  };
+});
+// mock the ReactWidget
+jest.mock('@jupyterlab/apputils', () => {
+  return {
+    __esModule: true,
+    ReactWidget: jest.fn().mockImplementation(() => {
+      return {
+        render: () => {
+          console.log('render');
+        }
+      };
+    })
+  };
+});
+
+import { PyunicoreComponent } from '../pyunicoreWidget';
+import {
+  findByText,
+  fireEvent,
+  getByTestId,
+  render,
+  waitFor
+} from '@testing-library/react';
+
+import {
+  BUTTON_SETTINGS,
+  getKernel,
+  mockGetJob
+} from './UnicoreJobsTable.test';
+import React from 'react';
+
+const data = {
+  message: '',
+  jobs: [
+    {
+      id: 'test1',
+      name: 'test_name_1',
+      owner: 'test',
+      site: 'JUDAC',
+      status: 'running',
+      resource_url: 'test_url',
+      start_time: '2022-02-18T10:54:08+0100',
+      is_cancelable: true,
+      logs: ['line 1', 'line 2']
+    },
+    {
+      id: 'test2',
+      name: 'test_name_2',
+      owner: 'test',
+      site: 'JUDAC',
+      status: 'FAILED',
+      resource_url: 'test_url_2',
+      start_time: '2022-02-18T10:54:08+0100',
+      is_cancelable: false,
+      logs: ['line 1', 'line 2']
+    }
+  ]
+};
+// add more jobs to have next button visible
+for (let i = 0; i < 10; i++) {
```

Comment:
Can't the data be generated in a single step here? 
Also, generating the dict with job details can be moved in a single place and called from all places necessary.

Reply:
Moved test data generation in a function to be called where needed.